src/Editor/ImGui/ImGuizmo.cpp

```diff
@@ -0,0 +1,1719 @@
+    static const float quadMin = 0.5f;
+    static const float quadMax = 0.8f;
+    static const float quadUV[8] = { quadMin, quadMin, quadMin, quadMax, quadMax, quadMax, quadMax, quadMin };
+    static const int halfCircleSegmentCount = 64;
+    static const float snapTension = 0.5f;
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    // 
+    static int GetMoveType(vec_t *gizmoHitProportion);
+    static int GetRotateType();
+    static int GetScaleType();
+
+    static ImVec2 worldToPos(const vec_t& worldPos, const matrix_t& mat)
+    {
+        ImGuiIO& io = ImGui::GetIO();
+
+        vec_t trans;
+        trans.TransformPoint(worldPos, mat);
+        trans *= 0.5f / trans.w;
+        trans += makeVect(0.5f, 0.5f);
+        trans.y = 1.f - trans.y;
+        trans.x *= gContext.mWidth;
+        trans.y *= gContext.mHeight;
+        trans.x += gContext.mX;
+        trans.y += gContext.mY;
+        return ImVec2(trans.x, trans.y);
+    }
+
+    static void ComputeCameraRay(vec_t &rayOrigin, vec_t &rayDir)
+    {
+        ImGuiIO& io = ImGui::GetIO();
+
+        matrix_t mViewProjInverse;
+        mViewProjInverse.Inverse(gContext.mViewMat * gContext.mProjectionMat);
+
+        float mox = ((io.MousePos.x - gContext.mX) / gContext.mWidth) * 2.f - 1.f;
+        float moy = (1.f - ((io.MousePos.y - gContext.mY) / gContext.mHeight)) * 2.f - 1.f;
+
+        rayOrigin.Transform(makeVect(mox, moy, 0.f, 1.f), mViewProjInverse);
+        rayOrigin *= 1.f / rayOrigin.w;
+        vec_t rayEnd;
+        rayEnd.Transform(makeVect(mox, moy, 1.f, 1.f), mViewProjInverse);
+        rayEnd *= 1.f / rayEnd.w;
+        rayDir = Normalized(rayEnd - rayOrigin);
+    }
+
+    static float IntersectRayPlane(const vec_t & rOrigin, const vec_t& rVector, const vec_t& plan)
+    {
+        float numer = plan.Dot3(rOrigin) - plan.w;
+        float denom = plan.Dot3(rVector);
+
+        if (fabsf(denom) < FLT_EPSILON)  // normal is orthogonal to vector, cant intersect
+            return -1.0f;
+
+        return -(numer / denom);
+    }
+
+    void SetRect(float x, float y, float width, float height)
+    {
+        gContext.mX = x;
+        gContext.mY = y;
+        gContext.mWidth = width;
+        gContext.mHeight = height;
+    }
+
+    void BeginFrame()
+    {
+        ImGuiIO& io = ImGui::GetIO();
+
+        ImGui::Begin("gizmo", NULL, io.DisplaySize, 0, ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoScrollbar | ImGuiWindowFlags_NoInputs | ImGuiWindowFlags_NoSavedSettings | ImGuiWindowFlags_NoFocusOnAppearing | ImGuiWindowFlags_NoBringToFrontOnFocus);
+
+        gContext.mDrawList = ImGui::GetWindowDrawList();
+
+        ImGui::End();
+    }
+
+    bool IsUsing()
+    {
+        return gContext.mbUsing || gContext.mbUsingBounds;
+    }
+
+    bool IsOver()
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'IsOver' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)